lib/time/time_test.go

```diff
@@ -0,0 +1,82 @@
+package time
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"go.starlark.net/starlark"
+)
+
+func TestPerThreadNowReturnsCorrectTime(t *testing.T) {
+	th := &starlark.Thread{}
+	date := time.Date(1, 2, 3, 4, 5, 6, 7, time.UTC)
+	SetNow(th, func() (time.Time, error) {
```

Comment:
We should defer a second call to SetNow to restore the old value of the global variable, to avoid perturbing other tests.

Reply:
But this is per-thread and we throw that thread away anyway?

Reply:
Of course. Never mind!